src/bridge.ts

```diff
@@ -998,12 +983,23 @@ export class Bridge {
             return existingIntent.intent;
         }
 
-        const client = this.clientFactory.getClientAs(userId, request);
+        const client = this.clientFactory.getClientAs(userId, request, !!this.opts.bridgeEncryption);
         const clientIntentOpts: IntentOpts = {
             backingStore: this.intentBackingStore,
             ...this.opts.intentOptions?.clients,
         };
         clientIntentOpts.registered = this.membershipCache.isUserRegistered(userId);
+        const encryptionOpts = this.opts.bridgeEncryption;
+        if (encryptionOpts) {
+            clientIntentOpts.encryption = {
+                sessionPromise: encryptionOpts.store.getStoredSession(userId),
```

Comment:
I guess here lies the answer why `sessionPromise` is a promise and not directly the session value..
because we can't await the promise here and we want to forward any errors to the parent.

Reply:
Ah yes, see above.